### PR TITLE
Codechange: Use std::visit for ScriptDataVariant and ScriptText::Param

### PR DESCRIPTION
**Pull request by PeterN**
## Motivation / Problem

Manually checkiing variant with `std::holds_alternative` then `std::get`

## Description

Use `std::visit` instead.

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR touches english.txt or translations? Check the [guidelines](https://github.com/OpenTTD/OpenTTD/blob/master/docs/eints.md)
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, game_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

### CODE REVIEW

File: src/script/api/script_text.cpp [hunk 1]
```diff
@@ -198,13 +198,21 @@ void ScriptText::ParamCheck::Encode(std::back_insert_iterator<std::string> &outp
...
-		fmt::format_to(output, ":");
-		Utf8Encode(output, SCC_ENCODED);
-		fmt::format_to(output, "{:X}", std::get<ScriptTextRef>(*this->param)->string);
-	}
+
+	struct visitor {
+		std::back_insert_iterator<std::string> &output;
+
+		void operator()(const std::string &value) { fmt::format_to(this->output, ":\"{}\"", value); }
+		void operator()(const SQInteger &value) { fmt::format_to(this->output, ":\"{}\"", value); }
```
Comment (LordAro) [inline src/script/api/script_text.cpp]: seem to have lost the `:X` formatting for this one

Comment (PeterN) [reply src/script/api/script_text.cpp]: Oops!

Comment (glx22) [reply src/script/api/script_text.cpp]: and extra quotes too

File: src/script/script_instance.cpp [hunk 2]
```diff
@@ -613,52 +613,46 @@ bool ScriptInstance::IsPaused()
...
-	if (std::holds_alternative<SQBool>(value)) {
-		sq_pushbool(vm, std::get<SQBool>(value));
-		return true;
-	}
+	struct visitor {
+		HSQUIRRELVM vm;
+		ScriptData *data;
+
+		bool operator()(const SQInteger &value) { sq_pushinteger(this->vm, value); return true; }
+		bool operator()(const std::string &value) { sq_pushstring(this->vm, value); return true; }
```
Comment (glx22) [inline src/script/script_instance.cpp]: Hmm there was a `-1` too for this one.

**Review by rubidium42 — APPROVED** [review]

**Pull request merged**